starknet_py/tests/e2e/client/client_test.py

```diff
@@ -338,6 +393,49 @@ async def test_custom_session(map_contract, network):
     assert gateway_client2._feeder_gateway_client.session.closed is True
 
 
+@pytest.mark.asyncio
+async def test_custom_session_full_node_client(map_contract, network):
+    # We must access protected `_client` to test session
+    # pylint: disable=protected-access
+
+    session = ClientSession()
+
+    tx_hash = (
+        await (
+            await map_contract.functions["put"].invoke(
+                key=10, value=20, max_fee=MAX_FEE
+            )
+        ).wait_for_acceptance()
+    ).hash
+
+    full_node_client1 = FullNodeClient(
+        node_url=network + "/rpc", net=network, session=session
+    )
+    full_node_client2 = FullNodeClient(
+        node_url=network + "/rpc", net=network, session=session
+    )
+
+    assert full_node_client1._client.session is not None
+    assert full_node_client1._client.session == session
+    assert full_node_client1._client.session.closed is False
+    assert full_node_client2._client.session is not None
+    assert full_node_client2._client.session == session
+    assert full_node_client2._client.session.closed is False
+
+    gateway1_response = await full_node_client1.get_transaction_receipt(tx_hash=tx_hash)
+    gateway2_response = await full_node_client2.get_transaction_receipt(tx_hash=tx_hash)
+    assert gateway1_response == gateway2_response
+
+    assert full_node_client1._client.session.closed is False
+    assert full_node_client2._client.session.closed is False
+
+    await session.close()
+
+    assert full_node_client1._client.session.closed is True
+    assert full_node_client2._client.session.closed is True
```

Comment:
I'm not sure about that, FullNodeClient has `_client property`, Gateway has `_feeder_gateway_client`, so asserts still need to stay. Refactoring in order to extract the other parts isn't worth it IMO since we want to deprecate Gateway later on anyway.

Reply:
It's 40 lines of almost duplicate code, if we can get rid of it, we would want to :)

How about

```
internal_client = client._client if isinstance(client, FullNodeClient) else client._feeder_gateway_client
session = internal_client.session
```

or

```
internal_client = getattr(client, "_client", client._feeder_gateway_client)
```

though I'm unsure about the second one working, might need 2x getattr.